<h1>PALMERS PENGUIN</h1>
<hr>


In [1]:
import pandas as pd
import numpy as py
import warnings
warnings.filterwarnings('ignore')

In [2]:
pengu = pd.read_csv("penguins.csv")

In [3]:
pengu["sex"].fillna("Unknown", inplace=True)
pengu["body_mass_g"].fillna(pengu["body_mass_g"].mean(), inplace=True)
pengu.fillna(pengu.median(numeric_only=True), inplace=True)

In [4]:
pengu.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      344.000000     344.000000         344.000000   344.000000   
mean        43.925000      17.152035         200.892442  4201.754386   
std          5.443792       1.969060          14.023826   799.613058   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.275000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h2>QUESTIONS</h2>

<h3>1. What specie has the longest average flipper length?</h3>


In [5]:
pengu['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [6]:
pengu['flipper_length_mm'].mean()


np.float64(200.8924418604651)

In [7]:
avg_flipper = pengu.groupby('species')['flipper_length_mm'].mean()

In [8]:
avg_flipper

species
Adelie       190.000000
Chinstrap    195.823529
Gentoo       217.024194
Name: flipper_length_mm, dtype: float64

<h2>Insight No. 1</h2>
<p>As you can see the result, Gentoo has the longest flipper lenght that has 217.186992 average flipper length.</p>
<hr>

<h3>2. Which species has the highest average body mass per year?</h3>

In [9]:
avg_bodymass = pengu.groupby(["year", "species"])["body_mass_g"].mean().reset_index()
heaviest_species = avg_bodymass.loc[avg_bodymass.groupby("year")["body_mass_g"].idxmax()]
heaviest_species

year species  body_mass_g
2  2007  Gentoo  5070.588235
5  2008  Gentoo  5019.565217
8  2009  Gentoo  5119.358054

<h2>Insight No. 2</h2>
<p>As you can see the result, Gentoo had the highest average body mass in every year.</p>
<hr>

<h3>3. How many Adelie males in our dataset?</h3>


In [10]:
males = pengu.groupby(['species', 'sex']).size()

In [11]:
males

species    sex    
Adelie     Unknown     6
           female     73
           male       73
Chinstrap  female     34
           male       34
Gentoo     Unknown     5
           female     58
           male       61
dtype: int64

<h2>Insight No. 3</h2>
<p>As you can see the result, we have 73 Adelie males in our dataset. </p>
<hr>

<h3>4. What specie has the shortest average bill depth in our dataset?</h3>


In [12]:
short = pengu.groupby('species')['bill_depth_mm'].mean()

In [13]:
short

species
Adelie       18.339474
Chinstrap    18.420588
Gentoo       15.000806
Name: bill_depth_mm, dtype: float64

<h2>Insight No. 4</h2>
<p>As you can see the result, among all specie, Adelie has the shortest bill depth which has the average of 18.346358 mm.  </p>
<hr>

<h3>5. Is there a correlation between bill depth and flipper length?</h3>


In [14]:
corr_bill_flip = pengu['bill_depth_mm'].corr(pengu['flipper_length_mm'])

In [15]:
corr_bill_flip

np.float64(-0.5838319610240155)

<h2>Insight No. 5</h2>
<p>As you can see, the result is -0.5838512164654122 meaning it has correlation between bill depth and flipper length   </p>
<hr>

<h3>6. What is the average bill length of Adelie penguin on Dream Island?</h3>

In [16]:
pengu["species"] == "Adelie"

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [17]:
pengu[pengu["island"] == "Dream"]["bill_length_mm"]


30     39.5
31     37.2
32     39.5
33     40.9
34     36.4
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 124, dtype: float64

In [18]:
adelie_dream_avg = pengu[(pengu["species"] == "Adelie") & (pengu["island"] == "Dream")]["bill_length_mm"].mean()


In [19]:
adelie_dream_avg

np.float64(38.50178571428571)

<h2>Insight No. 6</h2>
<p>As you can see, the average of Adelie's bill length in Dream island is 38.50178571428571 mm. </p>
<hr>

<h3>7. How many penguins lived in Dream Island?</h3>

In [20]:
dream_count = (pengu['island'] == 'Dream').sum()

In [21]:
dream_count

np.int64(124)

<h2>Insight No. 7</h2>
<p>As you can see, the result shows that we have 124 penguins lived in Dream Island </p>
<hr>

<h3>8. What is the average of bill length of Adelie penguins on Dream Island?</h3>

In [22]:
pengu["species"] == "Adelie"

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [23]:
pengu["island"] == "Dream"

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [24]:
ave_bill_length = pengu[(pengu["species"] == "Adelie") & (pengu["island"] == "Dream")]["bill_length_mm"].mean()

In [25]:
ave_bill_length

np.float64(38.50178571428571)

<h2>Insight No. 8</h2>
<p>As you can see, the average bill length of Adelie penguins on Dream Island is 38.519999999999996 mm. </p>
<hr>

<h3>9. On what island has the highest average body mass?</h3>

In [26]:
avg_bodymass_island = pengu.groupby("island")["body_mass_g"].mean()

In [27]:
avg_bodymass_island

island
Biscoe       4712.956871
Dream        3712.903226
Torgersen    3715.899123
Name: body_mass_g, dtype: float64

<h2>Insight No. 9</h2>
<p>As you can see, the island that has the highest average of body mass is Biscoe Island. </p>
<hr>

<h3>10. What is the minimum bill length per island?</h3>

In [28]:
bill_min = pengu.groupby("island")["bill_length_mm"].agg("min")

In [29]:
bill_min

island
Biscoe       34.5
Dream        32.1
Torgersen    33.5
Name: bill_length_mm, dtype: float64

<h2>Insight No. 10</h2>
<p>As you can see, the minimum bill length in Torgersen is 33.5, in Biscoe it is 34.5, and in Dream it is 32.1. </p>
<hr>

<h3>11. Does the bill depth of penguin have a effect in their body mass?</h3>

In [30]:
pengu["bill_depth_mm"].corr(pengu["body_mass_g"])


np.float64(-0.4719078065820587)

<h2>Insight No. 11</h2>
<p>As you can see, the result is -0.47191562118606645, meaning the bill depth of penguin has effect in their body mass. </p>
<hr>

<h3>12. Which year (2007, 2008, 2009) recorded the most penguins?</h3>

In [31]:
yr_count = pengu['year'].value_counts(ascending = False)

In [32]:
yr_count

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h2>Insight No. 12</h2>
<p>As you can see, year 2009 recorded the most penguins in our dataset. </p>
<hr>